src/mutation_data.cpp

```diff
@@ -387,6 +399,209 @@ void mutation_branch::reset_all()
 {
     mutations_category.clear();
     mutation_data.clear();
+    for ( auto &trait : trait_groups ) {
+        delete trait.second;
+    }
+    trait_blacklist.clear();
+    trait_groups.clear();
+    trait_groups["EMPTY_GROUP"] = new Trait_group_collection(100);
+}
+
+void mutation_branch::load_trait_blacklist( JsonObject &jsobj ) {
+    JsonArray jarr = jsobj.get_array( "traits" );
+    while (jarr.has_more()) {
+        trait_id id(jarr.next_string());
+        trait_blacklist.insert(id);
```

Comment:
I'm nitpicking now. This could be done in one line, without the `id` variable: `insert( trait_id( next_string() ) )`.